pkg/proxy/ipvs/proxier.go

```diff
@@ -2148,10 +2156,10 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 	return nil
 }
 
-func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, currentServices map[string]*utilipvs.VirtualServer, legacyBindAddrs map[string]bool) {
+func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, legacyBindAddrs map[string]bool) {
 	isIPv6 := utilnet.IsIPv6(proxier.nodeIP)
-	for cs := range currentServices {
-		svc := currentServices[cs]
+	for cs := range proxier.currentIPVSServices {
```

Comment:
Debating if maybe there should be a separate RWMutex for `currentIPVSServices` :thinking:

Reply:
I don't think it would be useful to have a separated mutex for `currentIPVSServices` since `Proxier.mu` is held during `syncProxyRules`. In my opinion it would take a larger refactor of `syncProxyRules` to see any benefit from splitting the `mu` mutex into multiple mutexes.